### PR TITLE
EEP: JSON Column Type

### PR DESCRIPTION
**Pull request by dmikurube**
This is the EEP (Embulk Enhancement Proposal) for the historical and the new JSON Column Type `org.embulk.spi.json.JsonValue`.

### CODE REVIEW

**Review by hiroyuki-sato — APPROVED** [review]
LGTM 👍 I left a little comment.

File: docs/eeps/draft-json-column-type.md [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+------------------------------
+
+One big consideration is JSON numbers. The JSON specification does not limit the range and precision of numbers, but at the same time, it also allows implementations to set limits. See [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259#section-6).
+
+From Embulk's viewpoint, Embulk has had the `LONG` and `DOUBLE` coulmn types since its beginning, but no column types representing unlimited numbers. Unlimited numbers in JSON could cause another problem about interoperability with the existing `LONG` and `DOUBLE` columns. For example, [`embulk-filter-expand_json`](https://github.com/embulk/embulk-filter-expand_json) expands JSON fields into Embulk columns. How would it expand a big JSON integer into a `LONG` column?  It would be reasonable to limit its number representations to Java primitive `long` and `double`.
+
+That being said, users sometimes want such a thing in the real world. Indeed, Embulk has received [such a request](https://github.com/embulk/embulk/issues/775) in the past. The new JSON values would leave some room for future extension with `BigInteger` and `BigDecimal`, such as `JsonBigInteger` and `JsonBigDecimal`.
```
Comment (civitaspo) [inline docs/eeps/draft-json-column-type.md:87]: nice:)

**Review by civitaspo — APPROVED** [review]

File: docs/eeps/draft-json-column-type.md [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+
+Last, the new JSON classes would not expect arbitrary extension by plugin developers. In other words, even if a plugin developer defines a new class implementing `JsonValue`, such as `MyJsonBigDecimal`, it would not work as the developer expects.
+
+`BigDecimal` and `BigInteger`
+------------------------------
+
+One big consideration is JSON numbers. The JSON specification does not limit the range and precision of numbers, but at the same time, it also allows implementations to set limits. See [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259#section-6).
+
+From Embulk's viewpoint, Embulk has had the `LONG` and `DOUBLE` coulmn types since its beginning, but no column types representing unlimited numbers. Unlimited numbers in JSON could cause another problem about interoperability with the existing `LONG` and `DOUBLE` columns. For example, [`embulk-filter-expand_json`](https://github.com/embulk/embulk-filter-expand_json) expands JSON fields into Embulk columns. How would it expand a big JSON integer into a `LONG` column?  It would be reasonable to limit its number representations to Java primitive `long` and `double`.
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply docs/eeps/draft-json-column-type.md:85]: Thanks for your comment! That's very specific about `double`, then that's described in "final class `JsonDouble`".

**Review by vietnguyen-td — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Thanks, all! Let's merge it as EEP-2 as `Accepted` along with #1556.

**Pull request merged**